docs/reinterpreting_effects.md

```diff
@@ -0,0 +1,359 @@
+# Reinterpreting effects
+
+This article assumes you are already familiar with [defining effects and their handlers](defining_effects.md).
+
+One of the nice aspects of effects is that they can support multiple effect handlers. Effects only specify actions, they don't actually perform them. Therefore, it's possible to "reinterpret" effects. There are multiple senses in which an effect can be reinterpreted:
+
+- Implementing an effect in terms of other effects. "Stacking" effects is a powerful tool for cleanly dividing implementations into the relevant abstraction layers with minimal leakage of implementation details.
+- Rewriting an effect and/or performing actions with the effect value and then performing the originally intended effect. This technique is conceptually similar to the middleware pattern commonly used in web applications.
```

Comment:
1. We (and others) usually refer to this as “interposition,” following Oleg Kiselyov et al’s work on extensible effects.

2. Middleware is an excellent point of comparison ✨ I’ve typically thought of aspect-oriented programming, but that’s much less likely to be familiar, so I’mma borrow this example in future 😊